### PR TITLE
Expr context

### PR DESCRIPTION
**Pull request by s-cork**
@albertjan - decide which commit you prefer of the penultimate and final commit 🤷

### CODE REVIEW

**Comment by albertjan** [comment]
`EXTRA_EXPR` is what cpython does right?

**Comment by s-cork** [comment]
yeah they use EXTRA_EXPR defined as a macro

**Comment by albertjan** [comment]
lets keep it, if someone else comes back to this in a few years and they're copying something from cpython, they won't have to go and find out why we don't have `EXTRA_EXPR` and what it all means. 😄 

but it looks nicer without. 😄

File: src/parser/pegen_types.ts [hunk 1]
```diff
@@ -93,3 +93,8 @@ export const FUNC_TYPE_INPUT = 345;
 export const FSTRING_INPUT = 800;
 
 export type StartRule = 256 | 257 | 258 | 345 | 800;
+
+export function EXTRA_EXPR(head: expr, tail?: expr): Attrs {
+    tail ??= head;
```
Comment (albertjan) [inline src/parser/pegen_types.ts:98]: what's this magic?

Comment (s-cork) [reply src/parser/pegen_types.ts:98]: if tail is null set it to the head
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Logical_nullish_assignment

Comment (s-cork) [reply src/parser/pegen_types.ts:98]: I can also just change this to make the second argument required like the cpython macro does 🤷‍♂️

Comment (albertjan) [reply src/parser/pegen_types.ts:98]: ohh cool

File: src/parser/pegen.ts [hunk 2]
```diff
@@ -1762,30 +1752,58 @@ function _set_name_context(e: Name, ctx: expr_context) {
 //     return _Py_Starred(_PyPegen_set_expr_context(p, e->v.Starred.value, ctx), ctx, EXTRA_EXPR(e, e));
 // }
 
-/* Creates an `expr_ty` equivalent to `expr` but with `ctx` as context */
+/* Creates an asdl_seq* where all the elements have been changed to have ctx as context */
+function _set_seq_context(p: Parser, seq: expr[], ctx: expr_context) {
```
Comment (albertjan) [inline src/parser/pegen.ts:1756]: we should add return types here

Comment (albertjan) [reply src/parser/pegen.ts:1756]: Or did you omit on purpose?

Comment (s-cork) [reply src/parser/pegen.ts:1760]: Will do. Sometimes I forget because it doesn't complain

**Pull request merged**